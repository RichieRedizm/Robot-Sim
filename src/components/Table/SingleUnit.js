import {
  faChevronDown,
  faChevronLeft,
  faChevronRight,
  faChevronUp,
  faRobot,
} from '@fortawesome/free-solid-svg-icons'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import React from 'react'

const SingleUnit = ({ unitId, classes }) => {
  return (
    <div data-unit-position={unitId} className={classes}>
      <FontAwesomeIcon icon={faChevronLeft} />
      <FontAwesomeIcon icon={faChevronUp} />
      <FontAwesomeIcon icon={faChevronRight} />
      <FontAwesomeIcon icon={faChevronDown} />
      <FontAwesomeIcon icon={faRobot} />
    </div>
  )
}

export default SingleUnit
