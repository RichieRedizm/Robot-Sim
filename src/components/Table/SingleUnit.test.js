import { cleanup, getByTestId, render } from '@testing-library/react'
import React from 'react'
import SingleUnit from './SingleUnit'

afterEach(cleanup)

const renderSingleUnitComponent = (props) => render(<SingleUnit {...props} />)

describe('SingleUnit component loads and renders correctly depending on props', () => {
  test('check if addRobot false effects the render', () => {
    const props = {
      unitId: 'x1y1',
      classes: 'unit-item',
      addRobot: false,
    }
    const { container, asFragment } = renderSingleUnitComponent(props)
    const unitItem = getByTestId(container, 'unitItem')
    expect(unitItem)
    // if addRobot is false the chevron and robot icons are not loaded for that unit
    expect(unitItem.children.length).toBe(0)
    expect(asFragment()).toMatchSnapshot()
  })
  test('check if addRobot false effects the render', () => {
    const props = {
      unitId: 'x1y1',
      classes: 'unit-item',
      addRobot: true,
    }
    const { container } = renderSingleUnitComponent(props)
    const unitItem = getByTestId(container, 'unitItem')
    // if addRobot is true the chevron and robot icons are loaded for that unit
    expect(unitItem.children.length).toBe(5)
  })
})
