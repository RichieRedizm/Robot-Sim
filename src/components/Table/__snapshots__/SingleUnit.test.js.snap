// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SingleUnit component loads and renders correctly depending on props check if addRobot false effects the render 1`] = `
<DocumentFragment>
  <div
    class="unit-item"
    data-testid="unitItem"
    data-unit-position="x1y1"
  />
</DocumentFragment>
`;
